include/seastar/core/abort_source.hh

```diff
@@ -185,6 +185,11 @@ public:
         }
     }
 
+    const std::exception_ptr& get_exception_ptr() const noexcept {
```

Comment:
1. the name is not entirely clear to the user IMO
2. Need to add doxygen comments to document the API

How about calling it `abort_requested_exception_ptr`?
This way it's clearer that it provides the same logical answer as `abort_requested`, just returning a different type, with additional information about the abort exception.

A hackish way might be to change `abort_requested` to return the exception_ptr instead of converting it to a `bool`, but let's be conservative and add a method instead.